test/unit/org/apache/cassandra/index/sai/cql/LuceneUpdateDeleteTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import org.junit.Test;
+
+import org.apache.cassandra.index.sai.SAITester;
+
+import static org.junit.Assert.assertEquals;
+
+public class LuceneUpdateDeleteTest extends SAITester
+{
+    @Test
+    public void removeUpdateAndDeleteTextInMemoryTest() throws Throwable
```

Comment:
this is a good starting point

(1) also want to mix in flushes -- do we get the same results on disk as in memory?  how about mixing them up where data is on disk/tombstone is in memory, or vice versa? look for calls to flush in the Vector tests for examples

(2) i'd like at least one test that does hundreds or thousands of writes -- okay for this to be a Long test if that makes more sense.  Mentioned where to find an example in the "field manual" doc.

Reply:
I just pushed up a commit satisfying the first request. I'll work on the hundreds/thousands of writes next.